engine/runtime/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -322,7 +322,8 @@ class Compiler(
         val moduleContext = ModuleContext(
           module          = module,
           freshNameSupply = Some(freshNameSupply),
-          compilerConfig  = config
+          compilerConfig  = config,
+          compiler        = Some(this)
```

Comment:
Now `ModuleContext` needs a compiler. I see it as a cyclic dependency and I am afraid of it.

Reply:
Yes, changed that to `PackageRepository` since we no longer trigger compilation. This is not ideal but due to the way passes and metadata are modeled, passes **have to** be defined as `case object` rather than `case class` and additional info has to be passed via context.

Ideally we would rewrite that since in situations like this it feels awkward but it's a much bigger task.